code/datums/weather/weather_types/rain_storm.dm

```diff
@@ -0,0 +1,153 @@
+	telegraph_overlay = "rain_low"
+	telegraph_duration = 30 SECONDS
+
+	weather_message = span_userdanger("<i>Rain pours down around you!</i>")
+	weather_overlay = "rain_high"
+
+	end_message = span_bolddanger("The downpour gradually slows to a light shower.")
+	end_overlay = "rain_low"
+	end_duration = 30 SECONDS
+
+	weather_duration_lower = 3 MINUTES
+	weather_duration_upper = 5 MINUTES
+
+	weather_color = null
+	thunder_color = null
+
+	area_type = /area
+	target_trait = ZTRAIT_RAINSTORM
+	immunity_type = TRAIT_RAINSTORM_IMMUNE
+	probability = 0
+	turf_weather_chance = 0.01
+	turf_thunder_chance = THUNDER_CHANCE_AVERAGE
+
+	weather_flags = (WEATHER_TURFS | WEATHER_MOBS | WEATHER_THUNDER | WEATHER_BAROMETER | WEATHER_NOTIFICATION)
+
+	/// A weighted list of possible reagents that will rain down from the sky.
+	/// Only one of these will be selected to be used as the reagent
+	var/list/whitelist_weather_reagents = list(/datum/reagent/water)
+	/// A list of reagents that are forbidden from being selected when there is no
+	/// whitelist and the reagents are randomized
+	var/list/blacklist_weather_reagents = list()
+	/// The selected reagent that will be rained down
+	var/datum/reagent/rain_reagent
+
+/datum/weather/rain_storm/New(z_levels, area_override, weather_flags_override, thunder_chance_override, datum/reagent/custom_reagent)
+	..()
+
+	if(IS_WEATHER_AESTHETIC(weather_flags))
+		return
+
+	var/reagent_id
+	if(custom_reagent)
+		reagent_id = custom_reagent
+	else if(whitelist_weather_reagents)
+		reagent_id = pick_weight_recursive(whitelist_weather_reagents)
+	else // randomized
+		reagent_id = get_random_reagent_id(blacklist_weather_reagents)
+
+	rain_reagent = find_reagent_object_from_type(reagent_id)
+
+	if(!rain_reagent)
+		CRASH("Attempted to call rain_storm weather with no rain_reagent present!")
+
+	weather_color = rain_reagent.color
+
+/datum/weather/rain_storm/telegraph()
+	setup_weather_areas(impacted_areas)
+
+	GLOB.rain_storm_sounds.Cut()
+	for(var/area/impacted_area as anything in impacted_areas)
+		GLOB.rain_storm_sounds[impacted_area] = /datum/looping_sound/rain/start
+		CHECK_TICK
+	return ..()
+
+/datum/weather/rain_storm/start()
+	GLOB.rain_storm_sounds.Cut()
+	for(var/area/impacted_area as anything in impacted_areas)
+		GLOB.rain_storm_sounds[impacted_area] = /datum/looping_sound/rain/middle
+		CHECK_TICK
+	return ..()
+
+/datum/weather/rain_storm/wind_down()
+	GLOB.rain_storm_sounds.Cut()
+	for(var/area/impacted_area as anything in impacted_areas)
+		GLOB.rain_storm_sounds[impacted_area] = /datum/looping_sound/rain/end
+		CHECK_TICK
+	return ..()
+
+/datum/weather/rain_storm/end()
+	GLOB.rain_storm_sounds.Cut()
+	return ..()
+
+/datum/weather/rain_storm/weather_act_mob(mob/living/living)
+	if(istype(rain_reagent, /datum/reagent/water))
+		living.wash()
+	rain_reagent.expose_mob(living, TOUCH, RAIN_REAGENT_VOLUME)
+
+	var/rain_type = LOWER_TEXT(rain_reagent.name)
+	if((weather_flags & WEATHER_NOTIFICATION) && prob(5))
```

Comment:
early return

Reply:
Yep

Reply:
Applied via 5975d7689f505e5f3beb7c360652c52e7a012941

Reply:
what happen to this commit?

Reply:
5975d7689f505e5f3beb7c360652c52e7a012941

Reply:
Oh wait. I see what happened. I copied the commit ID and posted the response here before pushing to remote. So Github didn't see the commit hash.